Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/ProximityChecker.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Informs about the proximity staus of the local player
```

Comment:
more comments on the netcode lesson here would be great for users.

Reply:
This applies to other Dioramas as well. We should make sure we explain things as much as possible

Reply:
Comments should be added to explain what isn't obvious. A property called `LocalPlayerIsClose` or `OnLocalPlayerProximityStatusChanged` don't need comments: their names are self-exaplaining.

the other prperties of this script have comments in the form of tooltips, which makes the explanations accessible even outside of code.

> This applies to other Dioramas as well. We should make sure we explain things as much as possible

Can you please point out the specific lines of code you think this feedback applies to?

Reply:
I'm not talking about what the code does, I'm talking about the underlying explanation. What's the multiplayer lesson people should be getting out of this? What are the challenges with netcode they should be careful about?
If you look at dynamic prefab's readme, we try to have those "here's the thing you should get out of this" for each section. For example, instead of just saying "here's an example for waiting for all clients to load an addressable before spawning", we say "some cases in your game like a boss character should be spawned only when all clients have finished loading, since this is critical to your gameplay and you want all clients to be able to see the boss before having do things in game. However, this means a client with trouble loading will block all other clients".
We're trying not to show just the how, but also the why, to go deeper in an explanation. This way users can apply this in their own games more easily.
That's the stuff we were talking about with Morwenna when talking about Bloom's taxonomy https://en.wikipedia.org/wiki/Bloom%27s_taxonomy Just showing the how is level 1 or 2. We want to reach level 3 or more if possible.

Reply:
> We want to reach level 3 or more if possible.

It'd have been nice to hear about this requirements when the PR was opened (3 months ago) or in the POI discussion we had.

I added more in-depth explanations in the latest commit to all relevant classes, please note that the bundled IET's also provide this type of teaching to some extent.